src/beliefpropagation/beliefpropagation.jl

```diff
@@ -99,7 +99,7 @@ function belief_propagation_iteration(
       ITensors.contract(ITensor(new_mts[src(e) => dst(e)]))
       LHS /= sum(diag(LHS))
       RHS /= sum(diag(RHS))
-      c += 0.5 * norm(LHS - RHS)
+      c += 0.5 * norm(dense(LHS) - dense(RHS))
```

Comment:
I guess there was a problem subtracting `Diag` tensors?

Also, `denseblocks` would be better to use since that works for block sparse tensors as well.

Reply:
Yeah there was. I have changed to use `denseblocks`